x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/integrations/install_promotion_rules.ts

```diff
@@ -0,0 +1,121 @@
+import { createPrebuiltRules } from '../rule_objects/create_prebuilt_rules';
+import type { IPrebuiltRuleObjectsClient } from '../rule_objects/prebuilt_rule_objects_client';
+import { upgradePrebuiltRules } from '../rule_objects/upgrade_prebuilt_rules';
+import type {
+  RuleBootstrapError,
+  RuleBootstrapResults,
+} from '../../../../../../common/api/detection_engine/prebuilt_rules/bootstrap_prebuilt_rules/bootstrap_prebuilt_rules.gen';
+import { getErrorMessage } from '../../../../../utils/error_helpers';
+import type { EndpointInternalFleetServicesInterface } from '../../../../../endpoint/services/fleet';
+import { AI_FOR_SOC_INTEGRATIONS, PROMOTION_RULE_TAG } from '../../../../../../common/constants';
+
+interface InstallPromotionRulesParams {
+  rulesClient: RulesClient;
+  detectionRulesClient: IDetectionRulesClient;
+  ruleAssetsClient: IPrebuiltRuleAssetsClient;
+  ruleObjectsClient: IPrebuiltRuleObjectsClient;
+  fleetServices: EndpointInternalFleetServicesInterface;
+}
+
+export async function installPromotionRules({
+  rulesClient,
+  detectionRulesClient,
+  ruleAssetsClient,
+  ruleObjectsClient,
+  fleetServices,
+}: InstallPromotionRulesParams): Promise<RuleBootstrapResults> {
+  // Get the list of installed integrations
+  const enabledIntegrations = new Set(
+    (
+      await Promise.all(
+        AI_FOR_SOC_INTEGRATIONS.map(async (integration) => {
+          const installation = await fleetServices.packages.getInstallation(integration);
+          return installation ? integration : [];
+        })
+      )
+    ).flat()
+  );
+
+  const latestRuleAssets = await ruleAssetsClient.fetchLatestAssets();
+
+  const latestPromotionRules = latestRuleAssets.filter((rule) => {
+    // Rule should be tagged as 'Promotion' and should be related to an enabled integration
+    return (
+      (rule.tags ?? []).includes(PROMOTION_RULE_TAG) &&
+      rule.related_integrations?.some((integration) => enabledIntegrations.has(integration.package))
+    );
+  });
+
+  const installedRuleVersions = await ruleObjectsClient.fetchInstalledRuleVersions();
+  const installedRuleVersionsMap = new Map(
+    installedRuleVersions.map((version) => [version.rule_id, version])
+  );
+
+  const promotionRulesToInstall = latestPromotionRules.filter(({ rule_id: ruleId }) => {
+    return !installedRuleVersionsMap.has(ruleId);
+  });
+  const { results: installationResults, errors: installationErrors } = await createPrebuiltRules(
+    detectionRulesClient,
+    promotionRulesToInstall.map((asset) => ({
+      ...asset,
+      enabled: true,
+    }))
+  );
+
+  const promotionRulesToUpgrade = latestPromotionRules.filter(({ rule_id: ruleId, version }) => {
+    const installedVersion = installedRuleVersionsMap.get(ruleId);
+    return installedVersion && installedVersion.version < version;
+  });
+  const { results: upgradeResults, errors: upgradeErrors } = await upgradePrebuiltRules(
```

Comment:
It looks like `upgradePrebuiltRules` will overwrite any exceptions, actions, or other modifications to the existing rule (although [exceptions and actions are saved](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/methods/upgrade_prebuilt_rule.ts#L60) if the rule type is being changed?). Do we plan to allow exceptions to be used with promotion rules? And is it a bug in `upgradePrebuiltRules` that rules changing types are treated differently than rules that don't change types in the upgrade?

Reply:
That's a good point. Rule actions and customizations won’t be supported, but I’m not sure yet if we’re going to use exceptions, I’ll check on that.